testing/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/problems/KnownProblemIds.groovy

```diff
@@ -63,15 +66,22 @@ class KnownProblemIds {
         'generic' : 'Generic'
     ]
 
-    private static final def KNOWN_DEFINITIONS = [
+    /**
+     * This map is used to validate that problems reported have known IDs, and display name.
+     * <p>
+     * Both the key and value is handled as a regular expression if the value is too dynamic.
+     */
+    private static final HashMap<String, String> KNOWN_DEFINITIONS = [
         'problems-api:missing-id' : 'Problem id must be specified',
         'problems-api:unsupported-additional-data' : 'Unsupported additional data type',
         'compilation:groovy-dsl:compilation-failed' : 'Groovy DSL script compilation problem',
-        'compilation:java:initialization-failed' : 'Java compilation initialization error',
-        'compilation:java:java-compilation-error' : 'Java compilation error',
-        'compilation:java:java-compilation-failed' : 'Java compilation error',
-        'compilation:java:java-compilation-warning' : 'Java compilation warning',
-        'compilation:java:java-compilation-advice' : 'Java compilation note',
+        // Flexible java compilation categories
+        // The end of the category is matched with a regex, as there are many possible endings (and also changes with JDK versions)
+        // See compiler.java for the full list of diagnostic codes we use as categories (we replace the dots with dashes)
+        'compilation:java:compiler-err-.+' : 'Java compilation error',
+        'compilation:java:compiler-warn-.+' : 'Java compilation warning',
+        'compilation:java:compiler-note-.+' : 'Java compilation note',
```

Comment:
This can probably go away.

Reply:
I don't think so: wouldn't this check fail if we start emitting problems with these IDs?

Reply:
ok